File: fix_bible.py
import re
import json

def fix_and_explore():
    path = 'c:/Users/david/Documents/quizz_biblique-/bible_lsg.json'
    with open(path, 'r', encoding='utf-8-sig') as f:
        content = f.read()

    # Quote keys: word followed by colon
    # This is a bit risky but let's try a simple regex for keys
    fixed = re.sub(r'(\w+):', r'"\1":', content)
    
    try:
        data = json.loads(fixed)
        print("Successfully converted to JSON")
        
        # Save fixed version for the web app to use
        with open('c:/Users/david/Documents/quizz_biblique-/bible_lsg_fixed.json', 'w', encoding='utf-8') as f:
            json.dump(data, f, ensure_ascii=False)
        print("Saved bible_lsg_fixed.json")
            
        book_names = []
        for testament in data['Testaments']:
            for book in testament['Books']:
                book_names.append(book.get('Text', 'Unknown'))
        
        print(f"Total Books: {len(book_names)}")
        print("Book names: " + ", ".join(book_names))
        
    except Exception as e:
        print(f"Error: {e}")
        # Print a snippet of where it failed
        if 'char' in str(e):
            pos = int(re.search(r'char (\d+)', str(e)).group(1))
            print(f"Snippet near error: {fixed[max(0, pos-50):pos+50]}")

if __name__ == "__main__":
    fix_and_explore()

File: fix_bible.js
const fs = require('fs');
const path = require('path');

const filePath = 'c:/Users/david/Documents/quizz_biblique-/bible_lsg.json';
const content = fs.readFileSync(filePath, 'utf8');

// The content is not valid JSON, but it looks like a JS object literal.
// We can try to evaluate it as an expression.
// Wrapping in parens to make it an expression.
try {
    const data = eval('(' + content + ')');
    const fixedPath = 'c:/Users/david/Documents/quizz_biblique-/bible_lsg_fixed.json';
    fs.writeFileSync(fixedPath, JSON.stringify(data, null, 2), 'utf8');
    console.log('Successfully saved valid JSON to bible_lsg_fixed.json');

    // Also log book names for my reference
    const bookNames = [];
    data.Testaments.forEach(t => {
        t.Books.forEach(b => {
            bookNames.push(b.Text);
        });
    });
    console.log('Total Books:', bookNames.length);
    console.log('Book Names:', bookNames.join(', '));
} catch (e) {
    console.error('Error evaluating or saving:', e);
}
